### PR TITLE
Add documentation related to the rasa helm chart

### PR DESCRIPTION
**Pull request by tczekajlo**
**Proposed changes**:
- Add documentation related to the rasa helm chart

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by tczekajlo** [comment]
@TyDunn @erohmensing I've added documentation related to the new helm chart for rasa open source. 

Please share your thoughts if you have a minute 🙏🏻  (as usual, I had a hard time with what name I should use, so you can find a mix of `Rasa Assistant/Rasa Open Source/...`) 🙈

**Comment by tczekajlo** [comment]
cc @tmbo

File: docs/docs/how-to-deploy.mdx
Comment (TyDunn) [inline docs/docs/how-to-deploy.mdx:167]:
```suggestion
We recommend installing the Rasa Open Source server in a separate
```

Comment (TyDunn) [reply docs/docs/how-to-deploy.mdx:167]: Or maybe just "Rasa Open Source" (without server?)

**Review by TyDunn — CHANGES_REQUESTED** [review]
Great work! We should revisit how / what we recommend re: deployment in the near term, but I think adding this as another option is the best way to proceed for now

File: docs/docs/how-to-deploy.mdx
Comment (tczekajlo) [reply docs/docs/how-to-deploy.mdx:167]: "Rasa Open Source" sounds good to me 😄

File: docs/docs/how-to-deploy.mdx
Comment (erohmensing) [inline docs/docs/how-to-deploy.mdx:198]: Let's decide consistently if we wnat to capitalize `rasa` in this
```suggestion
All available values you can find in [the Rasa helm chart repository](https://github.com/RasaHQ/helm-charts/tree/main/charts/rasa#values).

:::note
The default configuration of the Rasa chart deploys a Rasa Open Source Server, downloads a model, and serves the downloaded model.
Visit [the Rasa helm chart repository](https://github.com/RasaHQ/helm-charts/tree/main/charts/rasa#quick-start) to check out more examples of configuration.
```

File: docs/docs/how-to-deploy.mdx
Comment (erohmensing) [inline docs/docs/how-to-deploy.mdx:221]:
```suggestion
You can find an example that shows how to download data files from a git repository and train an initial model in the Rasa helm charts [examples](https://github.com/RasaHQ/helm-charts/blob/main/examples/rasa/train-model-helmfile.yaml).
```

File: docs/docs/how-to-deploy.mdx
Comment (erohmensing) [inline docs/docs/how-to-deploy.mdx:323]:
```suggestion
Visit [the Rasa helm chart repository](https://github.com/RasaHQ/helm-charts/tree/main/charts/rasa) where you can find examples of configuration and learn how to e.g. integrate your Rasa Open Source deployment with Rasa X.
```

File: docs/docs/how-to-deploy.mdx
Comment (erohmensing) [inline docs/docs/how-to-deploy.mdx:84]:
```suggestion
1. Check that you have installed the Kubernetes or OpenShift command line
```

File: docs/docs/how-to-deploy.mdx
Comment (erohmensing) [inline docs/docs/how-to-deploy.mdx:147]:
```suggestion
3. Make sure you have the [Helm CLI](https://helm.sh/docs/intro/install/)
```

**Review by erohmensing — COMMENTED** [review]
Not much to add, looks good 🏋🏽 I'll let Ty approve since he req'd changes

File: docs/docs/how-to-deploy.mdx
Comment (TyDunn) [inline docs/docs/how-to-deploy.mdx:59]:
```suggestion
by using the [Rasa Helm Chart](https://github.com/RasaHQ/helm-charts/tree/main/charts/rasa) in a scalable cluster environment using OpenShift or Kubernetes (K8S).
```

File: docs/docs/how-to-deploy.mdx
Comment (TyDunn) [inline docs/docs/how-to-deploy.mdx:73]:
```suggestion
The Rasa Helm chart is open source and available in the
```

**Review by TyDunn — APPROVED** [review]
Some small tweaks needed based on comments from Ella and me but LGTM! 🚀

**Pull request merged**
